GitUI/CommandsDialogs/FormCheckoutBranch.cs

```diff
@@ -267,13 +267,13 @@ private DialogResult OkClick()
                     IGitRef remoteBranchRef = GetRemoteBranchRef(cmd.BranchName);
                     if (localBranchRef != null && remoteBranchRef != null)
                     {
-                        string mergeBaseGuid = Module.GetMergeBase(localBranchRef.Guid, remoteBranchRef.Guid);
-                        bool isResetFastForward = localBranchRef.Guid.Equals(mergeBaseGuid);
+                        var mergeBaseGuid = Module.GetMergeBase(localBranchRef.Guid, remoteBranchRef.Guid);
+                        bool isResetFastForward = localBranchRef.Guid == mergeBaseGuid.ToString();
                         if (!isResetFastForward)
                         {
-                            string mergeBaseText = mergeBaseGuid.IsNullOrWhiteSpace()
+                            string mergeBaseText = mergeBaseGuid == null
                                 ? "merge base"
-                                : GitRevision.ToShortSha(mergeBaseGuid);
+                                : mergeBaseGuid.ToShortString();
```

Comment:
Can use the instance `ToShortString` instead of the static `GitRevision.ToShortSha` function. The output is the same.

Reply:
FYI: I've been reworking `GitRevision` for quite some time now to remove all behaviours from it. 
My refactor is not finished, my work and incoming PRs here blocked the progress.
The end `GitRevision` will be just a POCO